core/GlobalState.cc

```diff
@@ -2199,6 +2199,7 @@ unique_ptr<GlobalState> GlobalState::copyForIndex() const {
     result->onlyErrorClasses = this->onlyErrorClasses;
     result->suggestUnsafe = this->suggestUnsafe;
     result->pathPrefix = this->pathPrefix;
+    result->packageDB_ = this->packageDB_.emptyCopyWithOptions();
```

Comment:
I think this makes sense for consistency, but I'm a little surprised it's necessary: isn't `copyForIndex` only used in workers for `pipeline::index`? Do we gate behavior during indexing on `--stripe-packages` currently?

Reply:
I went back and forth on whether to include this here, because you're right, it's not required.

I think that ~soon, I am going to factor something in `GlobalState` called like `semanticOptionsHash` and use that when writing into the cache. I think that there are certain parts of rewriting `__package.rb` files that _would_ benefit from being moved to rewriter and cached (as far as simplifying the act of moving them into namer, and also for performance) and so I was anticipating that this might change soon, and figured better safe than sorry.

Reply:
That seems reasonable to me!